src/Features/Core/Portable/MapCode/IMapCodeService.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.MapCode
+{
+    /// <summary>
+    /// A service to identify, map, and integrate code snippets obtained from the AI model into a target document. 
+    /// A Mapper's primary responsibility is twofold:
+    /// 1. Replace existing code. The mapper should determine if the code provided by the model already exists in the target and needs to be replaced.
+    /// 2. Insert new code. The mapper should recognize new code and identify the correct location for its insertion in the target document.
+    /// </summary>
+    internal interface IMapCodeService : ILanguageService
+    {
+        /// <summary>
+        /// Map code snippets into the target document.
+        /// </summary>
+        /// <param name="document">Target document for mapping</param>
+        /// <param name="contents">Code snippet that we we are attempting to map into the target document</param>
```

Comment:
parameter indicates multiple contents, but you say "code snippet" (singular).

Note: do we need this to support multiple contents?  waht's the scenario for that?  Can it not just map a single content?